### PR TITLE
Pass `linear_operator` by value

### PR DESCRIPTION
**Pull request by umanwizard**
There's no reason AFAICT for this to be passed by `&mut` instead of just moved into the function.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/3962)

### CODE REVIEW

**Comment by benesch** [comment]
Are you sure about this? I'm only a bit familiar with this code, but I thought it was written such that the first component that applies the linear operator sets the linear operator to none, so that future operators in the pipeline don't need to bother.

**Comment by benesch** [comment]
I.e., I think you end up having to do work here? https://github.com/MaterializeInc/materialize/blob/main/src/dataflow/src/render/mod.rs#L482

Not super sure; this function is pretty hard to follow and I'm not very familiar with it!

**Comment by umanwizard** [comment]
That's in the `else` block (non-upsert case).

It's not possible that `src.operators` is being used after I move out of it here - otherwise, the compiler would fail with a use-after-move error.

**Comment by benesch** [comment]
> It's not possible that `src.operators` is being used after I move out of it here - otherwise, the compiler would fail with a use-after-move error.

That's a good point.

**Comment by benesch** [comment]
Welp, I'm going to tap out of this review because I don't think I have enough context. @wangandi landed approximately the inverse of this commit about a month ago (https://github.com/MaterializeInc/materialize/commit/2b2e77598276d55bda1dcaee0954c386632ba376), so not sure what to make of that.

**Comment by frankmcsherry** [comment]
I'm not sure about the state of the code at the moment (I would have thought this would have broken things) but the `&mut` reference existed initially so that source implementations could optionally implement whatever they are capable of implementing, and remove the part of the linear operator that they implemented.

**Comment by umanwizard** [comment]
Closing, because it seems this exists for a reason!
